sanic/blueprints.py

```diff
@@ -82,15 +83,18 @@ def decorator(handler):
             return handler
         return decorator
 
-    def add_route(self, handler, uri, methods=None, host=None):
+    def add_route(self, handler, uri, methods=frozenset({'GET'}), host=None):
         """
         Creates a blueprint route from a function.
-        :param handler: Function to handle uri request.
+        :param handler: Function or class instance to handle uri request.
         :param uri: Endpoint at which the route will be accessible.
         :param methods: List of acceptable HTTP methods.
+        :return: function or class instance
         """
-        route = FutureRoute(handler, uri, methods, host)
-        self.routes.append(route)
+        # Handle HTTPMethodView differently
+        if hasattr(handler, 'view_class'):
+            methods = frozenset(HTTP_METHODS)
```

Comment:
Should a view default to having all methods defined? Maybe this should just be `frozenset({'GET'})`?